nengo/builder/signal.py

```diff
@@ -69,22 +85,30 @@ def __init__(self, initial_value,
 
     def __getstate__(self):
         state = dict(self.__dict__)
-        v = self._initial_value
-        state['_initial_value'] = (
-            v.shape, v.base, npext.array_offset(v), v.strides)
+
+        if not self.sparse:
```

Comment:
Maybe make a note that we don't need this for sparse because we don't do views of sparse. If we do add sparse views at some point, this will have to change.

Reply:
Or we could do the note the other way, saying that we DO do this for normal arrays because they could be views.

Reply:
It's not clear to me that sparse views are possible... and I would assume if we're making a big change to how sparse signals are handled, we're going to do a search for `self.sparse` here and make changes as appropriate. So I'm not sure it warrants a specific comment right now.

Reply:
The other way round makes more sense, I'll add that comment

Reply:
As I mentioned in the slack channel, I was able to make two sparse matrices that share data (i.e. a sparse view of sorts). However, I think the method by which you create it would be different for each different sparse matrix subtype (i.e. CSR, COO, etc.), and we don't have an imminent need for it, so not worth pursuing now.

I just wanted to make it clear that this is happening because of views, since it would be easy to search for `self.sparse`, but completely ignore this code if you thought that it was something specific to sparse matrices, not to views. (This is why I wanted to keep the `viewable` property, to make this clearer, but a comment here about why we do this for dense will serve the same purpose.)